src/components/FlawAffects.vue

```diff
@@ -0,0 +1,1521 @@
+<script setup lang="ts">
+import { computed, toRefs, ref, type Ref } from 'vue';
+import {
+  affectImpacts,
+  affectAffectedness,
+  affectResolutions,
+  possibleAffectResolutions,
+} from '@/types/zodAffect';
+import { type ZodAffectType } from '@/types/zodAffect';
+import { uniques } from '@/utils/helpers';
+import { equals, clone, prop, descend, ascend, sortWith } from 'ramda';
+import FlawTrackers from '@/components/FlawTrackers.vue';
+import TrackerManager from '@/components/TrackerManager.vue';
+import LabelCollapsible from '@/components/widgets/LabelCollapsible.vue';
+import Modal from '@/components/widgets/Modal.vue';
+import { useModal } from '@/composables/useModal';
+import { useSettingsStore } from '@/stores/SettingsStore';
+import { usePagination } from '@/composables/usePagination';
+
+const settingsStore = useSettingsStore();
+const settings = ref(settingsStore.settings);
+const { isModalOpen, openModal, closeModal } = useModal();
+
+const props = defineProps<{
+  flawId: string;
+  embargoed: boolean;
+  affects: ZodAffectType[];
+  affectsToDelete: ZodAffectType[];
+  error: Record<string, any>[] | null;
+}>();
+
+const emit = defineEmits<{
+  'file-tracker': [value: object];
+  'affect:remove': [value: ZodAffectType];
+  'affect:recover': [value: ZodAffectType];
+  'affect:add': [value: ZodAffectType];
+  'affects:refresh': [];
+}>();
+
+const { affects, affectsToDelete } = toRefs(props);
+const hasAffects = computed(() => allAffects.value.length > 0);
+const allAffects = computed(() => affectsToDelete.value.concat(affects.value));
+const savedAffects = clone(affects.value) as ZodAffectType[];
+const affectsNotBeingDeleted = computed(
+  () => affects.value.filter((affect) => !affectsToDelete.value.includes(affect))
+);
```

Comment:
If we have to filter `affectsToDelete` from `affects` on line 44, does this means that `allAffects` has duplicated affects?

Reply:
Thanks for pointing out, been checking it and indeed `affects` is already without affects to delete cause in `useFlawAffectsModel` the affect removal method makes an splice on the affects array and adds it to the `affectsToDelete` so that filter was not actually filtering anything. 

Besides of using `affects` directly as `affectsNotBeingDeleted` prop to `FlawTrackers` I sightly changed the `displayedTrackers` computed property to filter there the `sortedAffects` to those not being deleted, so when an affect is set for removal the corresponding tracker is removed from the table.

Just noting that I don't use `affects` directly there cause `sortedAffects` has already filtered affects, so only trackers for the displayed affects are shown in the table.